llmnrd.c

```diff
@@ -196,6 +197,17 @@ int main(int argc, char **argv)
 
 	setlinebuf(stdout);
 
+	/* First count given (host)names, if any, to allocate memory */
+	while ((c = getopt_long(argc, argv, short_opts, long_opts, NULL)) != -1) {
+		if (c == 'H')
+			++name_count;
+	}
```

Comment:
I don't really like the idea of iterating the options twice. But I guess there is no alternative to this :-(